src/main/java/org/folio/edge/sip2/handlers/freemarker/FreemarkerUtils.java

```diff
@@ -0,0 +1,51 @@
+package org.folio.edge.sip2.handlers.freemarker;
+
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.lang.invoke.MethodHandles;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+public class FreemarkerUtils {
+
+  private static Logger log;
+
+  static {
+    log = LogManager.getLogger(MethodHandles.lookup().lookupClass());
+  }
+
+  /**
+   * Utility method that executes a freemarker template to return a templated-formmated string.
+   *
+   * @param data the data used by the template.
+   * @param template the template to apply to the data.
+   * @return String output of running the template on the data
+   */
+  public static String executeFreemarkerTemplate(Object data, Template template) {
```

Comment:
Maybe something to think about later, but we may want to decouple Freemarker with the actual business logic for building messages. Maybe have a "templating" abstraction and we happen to use Freemarker as the implementation, but we could replace it with some other templating system without changing the message building code.

Reply:
True! So much for my desire to have loosely coupled classes and I overlooked this part of the code!!